### DISCUSSION TITLE
Doesn't work

### DISCUSSION THREAD

**Discussion by arcibald22** [rxhanson/Middle-Localization#72] [Q&A]
Hi
I have 12.7 monterey 2,2 Ghz intel core i7 quad core.
When I tested the middleclick it worked, but when purchased license not more.
Why? Can you help me?
Only an information: if I do the download from APP STORE in application folder the name is middleclick, but if I use this download
https://middleclick.app/
the name in application is middle only. Is this a my error?
Regards

**Comment by rxhanson** [comment]
Thanks for purchasing. 

To figure out what's going on, I'll need some more information. If you open the about window in Middle and click Debug > Show Viewer, then we can see the trackpad or Magic Mouse touches that the app is seeing. When you execute a middle click gesture, the viewer will turn grey to show a middle click is being executed. Let me know what you see there.

I have two middle click apps: Middle (not on the App Store) and [MiddleTouch](https://middleclick.app/middletouch) (on the App Store). Other ones with similar names are not made by me.

**Comment by arcibald22** [comment]
Thank you for answer.
I purchased middletouch by applestore (red color). The FREE APP worked, the final not more.
Now I have MIDDLE (downloaded by your website, it's blu color); I don't use apple mouse, but normal PC mouse and I repeate with FREE APP worked.
If I do midddle click, no the viewer doesn't change.
After: I deleted blu app and installed by applestore the red color: the viewer has not reaction on all setup.
All the best.

**Reply by rxhanson** [reply]
Which gesture do you have selected? Perhaps you can take a screen recording with the viewer visible, so I can see what's going on?

**Comment by arcibald22** [comment]
Hi this is the link:
https://www.veoh.com/watch/v142329353Mbfb776g

The only time that you seen it worked is with left click.
Regards.

**Comment by rxhanson** [comment]
Thanks for the screen recording. I was hoping to see the touch viewer visible, like this:

https://github.com/rxhanson/Middle-Community/assets/13651296/5cbbf989-7ade-4dc9-a9ad-8269fcb39fb1

In that video, the first middle click is using a trackpad with the app, and the second one is using a mouse with a physical middle button, just for comparison.

The reason to provide the viewer in the video is so I can know:
1. Is the app receiving touch events from macOS?
2. If so, is there something strange about the placement of your touches on the trackpad? If you're resting your thumb on the edge of the trackpad then it will typically be rejected as a resting touch, and won't count as a touch when the gesture is interpreted (in case that's your expectation).

I suspect perhaps something is preventing the touch events from reaching the app, which is why I have the accessibility system settings panel displayed in my video showing that I have it enabled.

**Comment by arcibald22** [comment]
![track](https://github.com/rxhanson/Middle-Community/assets/35035518/7a4356b5-e28d-4198-ab5b-ab1a92f2a792)
Hi
setup it's correct.
This is my mouse, no external button:
https://www.amazon.it/Logitech-M185-Wireless-Mouse-Grigio/dp/B00552K0GM/ref=sr_1_5?dib=eyJ2IjoiMSJ9.YRFAB6nNtICBtfuJFxl-iWC9hCo3txo6t-w-FFso9YNLnz7BigZmByNhVBqsFGZKF37Z1tUNribRJbX9MPF5ZS2oRdeo0Rcfne6948t-PtTdtVtAF6jYbSNuAXEjL7Os3hMCRmUAGjUbYGHMxp6z9uCzKaUbOTIxHkTqb5BJK2WCCQSOP0gCNo_qvuYkCsgNCaorimpUJrqzPJl4Rujq5-Gr6Gpj4kHX45e2ZnAIhItal5xZMzh35waJTCqITRQDR5xNbb9A35ww5q_4uUJSWd9bu9uFvCcOQsvc3ux4vVI.UccQn4LKSDTaDuMoa7LKE1PZIZMfHroXgEfAysB49O0&dib_tag=se&keywords=mouse+wireless+logitech&qid=1713601751&sr=8-5
Regards

**Reply by rxhanson** [reply]
I think we're not on the same page here.

Are you trying to use MiddleTouch or Middle to middle click with your trackpad? MiddleTouch is for adding middle click functionality to an Apple trackpad, and Middle is for adding middle click functionality to an Apple Trackpad and/or Magic Mouse. They do not add middle click functionality to a 3rd party mouse.

**Comment by arcibald22** [comment]
The srange is that with the trail software and for the nxt 10 days with th epayed software all worked, after all out of the service.

**Comment by arcibald22** [comment]
HI RXHANSON a question: why the trial software worked perfectly and after 20 days i purchased the full software and it worked for 15 days very well?
Do you think I would buy the full version for what reason? Why didn't the trial version work? Thanks for the explanation to my strange behavior.
Best regards.

**Reply by rxhanson** [reply]
If you were experiencing a middle click on something other than an Apple Trackpad or Magic Mouse, then it cannot be attributed to MiddleTouch or Middle. I think if you're still within a certain timeframe, you can contact Apple's App Store support for a refund of your MiddleTouch purchase.
